### ISSUE TITLE
Doesn't correctly generate links to notes in folders

### ISSUE THREAD

**Issue by tiggerbiggo** [Pr0dt0s/obsidian-html-server#10]
Hi,

Vanilla Obsidian automatically finds where files are even in sub folders without needing to include the full path in [[links]], but the web server does not. For example [[something]] may point to /folder/to/something.md, however when clicking a link from /index.md, the web server tries to go to localhost:8080/something, which does not work. 

localhost:8080/path/to/something resolves correctly, so the problem is only with the generation of links.

Thanks very much for the plugin as it works if i just flatten my files to the root folder, hope this is an easy fix because I like organising into sub folders even though obsidian navigates them for me.

**Comment by tiggerbiggo** [comment]
Just adding some more details:

OS: Windows 10
Obsidian version: v1.3.7
Plugin version: 1.0.5

Thanks again :)

**Comment by Pr0dt0s** [comment]
Hello @tiggerbiggo, I'm going to look into this. This is the first time I hear about this functionality but that is actually how Obsidian works.

**Comment by hiyorijl** [comment]
yeah exactly I came to write down this issue Vanilla Obsidian can recognize folder without having to write down the full path.

### My usecase
I create folders on obsidian for tons of stuff like the folder `Daily Planners` (self-explanatory), I'm using the extension _changelog_ `badrbouslikhin/obsidian-vault-changelog` to put `changelog/changelog.md` as the _Index File_ in the config of `Html Server` of the clickable Obsidian Settings. So since I'm putting a folder reference in _Index File_ `Pr0dt0s/obsidian-html-server` indeed can work if directed correctly to the path of the folder.

My concrete example is the following I'm clicking on the changelog/changelog.md  which is the default of http://localhost:8080/  when I click on [[2023-08-14]] it directs me to http://localhost:8080/2023-08-14 (when I click into it, it directs me to a blank file, only appearing its title) instead of http://localhost:8080/Day%20Planners/2023-08-14

I hope what I wrote was understandable, forgive my English! Very nice extension btw, the default UX is so pretty and orderly-like 🥹.

**Comment by henusys** [comment]
This is a great plugin, but there are still some imperfections that I hope can be fixed.  

When I use the URL 'http://localhost:8080/2023-09-09', in the browser to open notes, it will show: 'Couldn' t resolve the file at path '/' 2023-09-09 '. But if I use the URL 'http://localhost:8080/DailyNote/2023-09-09' it will be correct.
Actually my file is in the path '/DailyNote/2023-09-09'.

The same problem exists with the attached files.

**Comment by Pr0dt0s** [comment]
Just published a new version with a fix for all these problems, please note that you must have `<base href="/">` in the head element in your `Custom index.html file` setting as in the image below.

![image](https://github.com/Pr0dt0s/obsidian-html-server/assets/24417072/0fe89180-1d4c-475d-a621-781b282be2c8)

You could just type (or copy and paste) it in your settings, or reset the value of the setting to the *new* default.

**Comment by Pr0dt0s** [comment]
I'll leave the issue open waiting for your feedback. Please let me know how it goes.

**Comment by henusys** [comment]
Yes, it well done! But I don't know if you have free time and enthusiasm to resolve the list of the problem:
1, Now, the function of Use Simple Authentication dosn't work.
2, Can it add the support for the render of Ecalidraw and dataview plugins?

**Comment by Pr0dt0s** [comment]
Hi @henusys, seems like the latest changes introduced that bug in the simple authentication. I just released a new version addressing that problem, should be woking again as intended.

About the support for the mentioned plugins I'm going to look into it, but I will track that progress in a new Issue.

**Comment by henusys** [comment]
That is a great job, and I just find you add two new issues.  
To make this plugin even better, I'll mention one more improvement, hopefully the html server formula rendering will be cons
istent with obsidian as shown below.
![1694424262224](https://github.com/Pr0dt0s/obsidian-html-server/assets/9565297/44c481a1-816b-4117-87dc-96dda74b9200)

**Comment by Pr0dt0s** [comment]
Hey @henusys, thanks a lot for the feedback! I'm really glad this software is of any use to at least someone.

Would you mind opening a new issue for that problem, and please detail which plugin you are using.
